### PR TITLE
Functionality to star and unstar modules

### PR DESCRIPTION
**Pull request by helloqx**
High Level Feature 9
* Can star/unstar modules from View Module page
* Can see starred modules from Starred Modules page
* Starred Modules page behaves as a truncated Module Listing page

### CODE REVIEW

File: test/test_star_listing.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
+'''
+    test_module_listing.py test the module listing view.
```
Comment (nlzz22) [inline test/test_star_listing.py]: Please rename the docstring for this file

File: test/test_star_listing.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+'''
+from paste.fixture import TestApp
+from nose.tools import assert_equal
+from app import APP
+from components import session, model
+
+
+class TestCode(object):
+    '''
+        This class runs the test cases to test app home page
```
Comment (nlzz22) [inline test/test_star_listing.py]: Please update the class's docstring

File: test/test_star_listing.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        response.mustcontain("Not Found")
+
+
+    def test_index_starredModules_listing(self):
+        '''
+            Tests if a table displaying a list of all starredModules exist.
+        '''
+        root = self.test_app.get('/starredModules')
+
+        # Checks the existence of the handler for viewing fixed mounting plan
```
Comment (nlzz22) [inline test/test_star_listing.py]: Please update this comment.

File: test/test_view_module.py [hunk 4]
```diff
@@ -227,3 +233,24 @@ def test_goto_edit_general_info(self):
...
+
+
+    def test_click_star_button(self):
+        '''
+            Tests if user can star and unstar modules by clicking the button.
+        '''
+        root = self.test_app.get(self.URL_VIEW_MODULE_VALID)
+        response = root.goto(self.URL_STAR_MODULE, method='get')
+        assert_equal(response.status, 303)
+        # follow back to view module page
```
Comment (nlzz22) [inline test/test_view_module.py]: please update the comment for lines 245 and 253

**Comment by helloqx** [comment]
Fixed as per requests @nlzz22 @xaterz

**Comment by xaterz** [comment]
Changes requested have been made. However there is a bug now in individualModuleInfo. The steps to replicate it are:
1) Go to http://localhost:9000/viewModule?code=BT5110
2) Click on 'AY 16/17 Sem 1'
3) Star the module. Module is shown as starred.
4) Press back button of your browser. Module is shown as unstarred.
5) Click on the star again. Psycopg2 IntegrityError occurs.

**Comment by helloqx** [comment]
@xaterz fixed and ready to merge
User still advised not to use back button tho 😂

**Comment by nlzz22** [comment]
image
@helloqx  Please take note that "Recently Removed Modules" is shown at the home page, which is not supposed to be shown.

**Pull request merged**